From 571bc633ee9257f1d74b28ff8151ac2cc148712b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Fr=C3=A9d=C3=A9ric=20Pierret=20=28fepitre=29?=
 <frederic.pierret@qubes-os.org>
Date: Fri, 9 Oct 2020 17:02:13 +0200
Subject: [PATCH] rpmdb2solve.c: add homedir functionality

Based on previous original work of author in https://lists.debian.org/debian-devel/2019/09/msg00218.html
---
 tools/rpmdb2solv.c | 11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/tools/rpmdb2solv.c b/tools/rpmdb2solv.c
index 1a04adf8..624be0da 100644
--- a/tools/rpmdb2solv.c
+++ b/tools/rpmdb2solv.c
@@ -41,8 +41,9 @@ static void
 usage(int status)
 {
   fprintf(stderr, "\nUsage:\n"
-	  "rpmdb2solv [-P] [-C] [-n] [-b <basefile>] [-p <productsdir>] [-r <root>]\n"
+	  "rpmdb2solv [-P] [-C] [-n] [H] [-b <basefile>] [-p <productsdir>] [-r <root>]\n"
 	  " -n : No packages, do not read rpmdb, useful to only parse products\n"
+	  " -H : use rpmdb in user's home directory\n"
 	  " -p <productsdir> : Scan <productsdir> for .prod files, representing installed products\n"
 	  " -r <root> : Prefix rpmdb path and <productsdir> with <root>\n"
 	  " -o <solv> : Write .solv to file instead of stdout\n"
@@ -63,6 +64,7 @@ main(int argc, char **argv)
   int c, percent = 0;
   int nopacks = 0;
   int add_changelog = 0;
+  int homedir = 0;
   const char *root = 0;
   const char *refname = 0;
 #ifdef ENABLE_SUSEREPO
@@ -83,7 +85,7 @@ main(int argc, char **argv)
    * parse arguments
    */
   
-  while ((c = getopt(argc, argv, "ACPhnkxXr:p:o:")) >= 0)
+  while ((c = getopt(argc, argv, "ACPhnHkxXr:p:o:")) >= 0)
     switch (c)
       {
       case 'h':
@@ -127,6 +129,9 @@ main(int argc, char **argv)
       case 'C':
 	add_changelog = 1;
 	break;
+      case 'H':
+        homedir = 1;
+        break;
       default:
 	usage(1);
       }
@@ -171,6 +176,8 @@ main(int argc, char **argv)
 	flags |= RPMDB_REPORT_PROGRESS;
       if (add_changelog)
 	flags |= RPM_ADD_WITH_CHANGELOG;
+      if (homedir)
+        flags |= RPMDB_USE_HOMEDIR;
       if (repo_add_rpmdb_reffp(repo, reffp, flags))
 	{
 	  fprintf(stderr, "rpmdb2solv: %s\n", pool_errstr(pool));
-- 
2.26.2

